### ISSUE TITLE
Poetry removing extras on add/update of individual package

### ISSUE THREAD

**Issue by jasongi-actu** [sdispater/poetry#830]
- [x] I am on the [latest](https://github.com/sdispater/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/sdispater/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: OSX Mojave
- **Poetry version**: 0.12.11
- **Link of a [Gist](https://gist.github.com/) with the contents of your pyproject.toml file**: See below

## Issue
When you have a nested dependency that has extras installed, and you add/update an unrelated package, it will uninstall the extra. An example is as follows and can be reproduced using the following script

```
poetry new test
poetry new test2
poetry new test3
cd test
poetry add fuzzywuzzy -E speedup
cd ../test2
poetry add test --path ../test
cd ../test3
poetry add test2 --path ../test2
poetry add pendulum
```

```
actu15mac18:test jgiancono$ poetry new test; poetry new test2; poetry new test3; cd test; poetry add fuzzywuzzy -E speedup; cd ../test2; poetry add test --path ../test; cd ../test3; poetry add test2 --path ../test2; poetry add pendulum;
Created package test in test
Created package test2 in test2
Created package test3 in test3
Using version ^0.17.0 for fuzzywuzzy

Updating dependencies
Resolving dependencies... (1.3s)


Package operations: 12 installs, 0 updates, 0 removals

Writing lock file

  - Installing scandir (1.9.0)
  - Installing six (1.12.0)
  - Installing atomicwrites (1.2.1)
  - Installing attrs (18.2.0)
  - Installing funcsigs (1.0.2)
  - Installing more-itertools (5.0.0)
  - Installing pathlib2 (2.3.3)
  - Installing pluggy (0.8.1)
  - Installing py (1.7.0)
  - Installing python-levenshtein (0.12.0)
  - Installing fuzzywuzzy (0.17.0)
  - Installing pytest (3.10.1)
Creating virtualenv test2-py2.7 in /Users/jgiancono/Library/Caches/pypoetry/virtualenvs

Updating dependencies
Resolving dependencies... (0.2s)


Package operations: 13 installs, 0 updates, 0 removals

Writing lock file

  - Installing python-levenshtein (0.12.0)
  - Installing scandir (1.9.0)
  - Installing six (1.12.0)
  - Installing atomicwrites (1.2.1)
  - Installing attrs (18.2.0)
  - Installing funcsigs (1.0.2)
  - Installing fuzzywuzzy (0.17.0)
  - Installing more-itertools (5.0.0)
  - Installing pathlib2 (2.3.3)
  - Installing pluggy (0.8.1)
  - Installing py (1.7.0)
  - Installing pytest (3.10.1)
  - Installing test (0.1.0 ../test)
Creating virtualenv test3-py2.7 in /Users/jgiancono/Library/Caches/pypoetry/virtualenvs

Updating dependencies
Resolving dependencies... (0.2s)


Package operations: 14 installs, 0 updates, 0 removals

Writing lock file

  - Installing python-levenshtein (0.12.0)
  - Installing fuzzywuzzy (0.17.0)
  - Installing scandir (1.9.0)
  - Installing six (1.12.0)
  - Installing atomicwrites (1.2.1)
  - Installing attrs (18.2.0)
  - Installing funcsigs (1.0.2)
  - Installing more-itertools (5.0.0)
  - Installing pathlib2 (2.3.3)
  - Installing pluggy (0.8.1)
  - Installing py (1.7.0)
  - Installing test (0.1.0 ../test2/../test)
  - Installing pytest (3.10.1)
  - Installing test2 (0.1.0 ../test2)
Using version ^2.0 for pendulum

Updating dependencies
Resolving dependencies... (1.0s)


Package operations: 4 installs, 0 updates, 1 removal

Writing lock file

  - Installing python-dateutil (2.7.5)
  - Installing pytzdata (2018.7)
  - Installing typing (3.6.6)
  - Installing pendulum (2.0.4)
  - Removing python-levenshtein (0.12.0)
actu15mac18:test3 jgiancono$ 
```
It removes python-levenshtein (which was installed by the extra "speedup" when it shouldn't do anything.

**Comment by sdispater** [comment]
This has normally been fixed in the 1.0.0 prereleases (the latest one is `1.0.0a2`). This can't be fixed in a bugfix release since it required some changes to the lock file.

Can you test and tell me if that fixes your issue?

**Comment by madig** [comment]
1.0.0a2 fixes the issue for me.
